managers/docker-manager/DockerService.js

```diff
@@ -0,0 +1,801 @@
+        logger.error(err);
+        throw err;
+      });
+  }
+
+  createAndStartContainer(opts, isNew) {
+    const self = this;
+
+    function attempt(tries) {
+      return Promise
+        .try(() => {
+          if (isNew && tries > 0) {
+            return self.createPortBindings(opts.ExposedPorts)
+              .tap(portBindings => _.set(opts.HostConfig, 'PortBindings', portBindings));
+          }
+        })
+        .then(() => self
+          .createContainer(opts)
+        )
+        .then(() => self
+          .startContainer()
+          .catch(DockerError.ServerError, err => self
+            .removeContainer()
+            .then(() => docker.updatePortRegistry())
+            .throw(new ContainerStartError(err.message))
+          )
+        );
+    }
+
+    function throwTimeoutError(err) {
+      throw err.error;
+    }
+
+    return utils
+      .retry(attempt, {
+        predicate: ContainerStartError,
+        maxAttempts: 3,
+        minDelay: 50
+      })
+      .catch(Timeout, throwTimeoutError)
+      .then(() => this.inspectContainer());
+  }
+
+  removeContainer() {
+    logger.info(`Removing docker container '${this.container.id}'...`);
+    return this.container
+      .removeAsync({
+        v: true,
+        force: true
+      })
+      .tap(() => logger.info(`+-> docker container has been removed with id '${this.container.id}' `))
+      .catch(err => {
+        logger.error(`+-> Failed to remove docker container with id '${this.container.id}' `);
+        logger.error(err);
+        throw err;
+      });
+  }
+
+  removeVolume(volumeName) {
+    logger.info(`Removing Docker volume '${volumeName}'...`);
+    return this.docker
+      .getVolume(volumeName)
+      .removeAsync()
+      .tap(() => logger.info(`+-> Docker volume has been removed with volume name - '${volumeName}' `))
+      .catch(DockerError.NotFound, () => logger.warn(`+-> Docker volume not found with volume name '${volumeName}' `))
+      .catch(err => {
+        logger.error(`+-> Failed to remove Docker volume with name '${volumeName}' `);
+        logger.error(err);
+        throw err;
+      });
+  }
+
+  removeVolumes() {
+    return this.docker
+      .listVolumesAsync({
+        filters: '{"dangling":{"true":true}}'
+      })
+      .then(result => _
+        .chain(result)
+        .get('Volumes')
+        .filter(volume => _.startsWith(volume.Name, this.containerName))
+        .value()
+      )
+      .map(volume => this.removeVolume(volume.Name))
+      .return();
+  }
+
+  getNetworkInfo(containerInfo) {
+    const info = {
+      ports: {}
+    };
+    _.each(containerInfo.NetworkSettings.Ports, (values, key) => {
+      if (_.size(values)) {
+        const value = _.first(values);
+        info.ip = value.HostIp;
+        info.ports[key] = value.HostPort;
+      }
+    });
+    if (info.ip === '0.0.0.0') {
+      info.ip = this.hostIp;
+    }
+    return info;
+  }
+
+  // getProcesses() {
```

Comment:
please dont checkin commented out code.

Reply:
These methods are copied from old Instance/Manager files. They are still used with the old flow as part of dashboard show, which should eventually be removed. Kept it here so that we can uncomment it here and remove from old Instance/Manager files. Also we decided not to touch the Instance/Manager files in the current PR. They also will be cleaned up.